gcloud/datastore/query.py

```diff
@@ -463,6 +463,11 @@ def __iter__(self):
                 yield entity
             if not self._more_results:
                 break
+            num_results = len(self._page)
+            if self._limit is not None:
+                self._limit -= num_results
+            if self._offset is not None:
+                self._offset += num_results
```

Comment:
Doesn't the offset get larger? Or you're saying it should be less since the cursor will inform the right location?